### PR TITLE
[SPARK-37554][BUILD] Add PyArrow, pandas and plotly to release Docker image dependencies

### PR DESCRIPTION
**Pull request by HyukjinKwon**
### What changes were proposed in this pull request?

This PR proposes to add plotly, pyarrow and pandas dependencies for generating the API documentation for pandas API on Spark.

The versions of `pandas==1.1.5 pyarrow==3.0.0 plotly==5.4.0` are matched with the current versions being used in branch-3.2 at Python 3.6.

### Why are the changes needed?

Currently, the function references for pandas API on Spark are all missing: https://spark.apache.org/docs/latest/api/python/reference/pyspark.pandas/series.html due to missing dependencies when building the docs.

### Does this PR introduce _any_ user-facing change?

Yes, the broken links of documentation at https://spark.apache.org/docs/latest/api/python/reference/pyspark.pandas/series.html will all be recovered.

### How was this patch tested?

To be honest, it has not been tested. I don't have the nerve to run Docker releasing script for the sake of testing so I defer to the next release manager.

The combinations of the dependency versions are being tested in GitHub Actions at `branch-3.2`.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
cc @gengliangwang and @dongjoon-hyun FYI

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Comment by HyukjinKwon** [comment]
Thanks @dongjoon-hyun. I am merging this in as CI won't verify this change in any event.

**Comment by HyukjinKwon** [comment]
Merged to master.

**Comment by dongjoon-hyun** [comment]
Sure, no problem.

BTW, cc @huaxingao since she is the next release manager.

**Comment by HyukjinKwon** [comment]
BTW, I am trying to resolve SPARK-37004 (3.2.1 blocker) as soon as possible. Once Py4J releases, we will resolve this by upgrading Py4J version soon. I asked the author to make a release, and he confirmed that he will work on that soon.

If that gets relayed like a week again, I will try to work around within PySpark to unblock the release 3.2.1.

**Comment by HyukjinKwon** [comment]
Oh, my mistake. He actually already did https://github.com/bartdag/py4j/tree/0.10.9.3. I will upgrade it in PySpark soon by EOD today.

**Comment by dongjoon-hyun** [comment]
I backported this to `branch-3.2` for Apache Spark 3.2.2 preparation.

**Comment by HyukjinKwon** [comment]
Thanks!!!

**Pull request closed without merging**
